services/ux-backend/main.go

```diff
@@ -58,11 +79,17 @@ func main() {
 	})
 
 	klog.Info("ux backend server listening on port ", config.listenPort)
+	if config.runServerWithTLS {
+		log.Fatal(http.ListenAndServeTLS(
+			fmt.Sprintf("%s%d", ":", config.listenPort),
+			"/etc/tls/private/tls.crt",
+			"/etc/tls/private/tls.key",
+			nil,
+		))
+	}
 
-	log.Fatal(http.ListenAndServeTLS(
+	log.Fatal(http.ListenAndServe(
```

Comment:
This should be in an `else`

Reply:
in an else I cannot write log.Fatel as it return function and if I do golint will cry

Reply:
Basically i might get an error something like this (if block ends with call to log.Fatal function, so drop this else and outdent its block)

Reply:
Ok. then seprate the fatal from the return value
```
	var err error
	addr := fmt.Sprintf("%s%d", ":", config.listenPort)
	if config.tlsEnabled {
		err := http.ListenAndServeTLS(addr,
			"/etc/tls/private/tls.crt",
			"/etc/tls/private/tls.key",
			nil,
		))
	} else {
		err := http.ListenAndServe(addr, nil)
	}
	log.Fatal(err)
```